server/routes/makeAReferral/completion-deadline/completionDeadlineForm.ts

```diff
@@ -1,20 +1,86 @@
+import { ReferralDetailsFormUpdate } from '../../../models/referralUpdate'
+import { FormValidationError } from '../../../utils/formValidationError'
+import FormUtils from '../../../utils/formUtils'
+import { FormValidationResult } from '../../../utils/forms/formValidationResult'
 
 export default class CompletionDeadlineForm {
-  constructor(private readonly request: Request) {}
+  constructor(private readonly request: Request, private readonly isSentReferral: boolean) {}
 
-  async data(): Promise<FormData<Partial<DraftReferral>>> {
+  static readonly completionDeadlineFieldId = 'completion-deadline'
+
+  static readonly reasonForChangeFieldId = 'reason-for-change'
+
+  async data(): Promise<FormData<ReferralDetailsFormUpdate>> {
     const input = new CalendarDayInput(this.request, 'completion-deadline', errorMessages.completionDeadline)
-    const result = await input.validate()
+    const completionDeadlineResult = await input.validate()
+    let reasonForChangeResult: FormValidationResult<string> | null = null
+
+    if (this.isSentReferral) {
+      reasonForChangeResult = await CompletionDeadlineForm.validate(this.request)
+
+      return completionDeadlineResult.error || reasonForChangeResult!.error
+        ? {
+            paramsForUpdate: null,
+            error: {
+              errors: [
+                ...(completionDeadlineResult.error?.errors ?? []),
+                ...(reasonForChangeResult!.error?.errors ?? []),
+              ],
+            },
+          }
+        : {
+            error: null,
+            paramsForUpdate: {
+              draftReferral: { completionDeadline: completionDeadlineResult.value.iso8601 },
+              reasonForChange: this.request.body[CompletionDeadlineForm.reasonForChangeFieldId],
+            },
+          }
+    }
+    return completionDeadlineResult.value
+      ? {
+          error: null,
+          paramsForUpdate: {
+            draftReferral: { completionDeadline: completionDeadlineResult.value.iso8601 },
+            reasonForChange: null,
+          },
+        }
+      : { error: completionDeadlineResult.error, paramsForUpdate: null }
+  }
 
-    if (result.value) {
-      return { error: null, paramsForUpdate: { completionDeadline: result.value.iso8601 } }
+  static get validations() {
+    return [
+      ExpressValidator.body(CompletionDeadlineForm.reasonForChangeFieldId)
+        .notEmpty({ ignore_whitespace: true })
+        .withMessage(errorMessages.completionDeadline.cannotBeEmpty),
+    ]
+  }
+
+  static async validate(request: Request): Promise<FormValidationResult<string>> {
+    const validationResult = await FormUtils.runValidations({
+      request,
+      validations: CompletionDeadlineForm.validations,
+    })
+
+    const error = CompletionDeadlineForm.extractErrors(validationResult)
+    if (error) {
+      return { value: null, error }
     }
+    return {
+      value: CompletionDeadlineForm.reasonForChangeFieldId,
```

Comment:
slightly confused about what this return value represents. should the value be the actual user-entered value in the form? how is this return value being used?

Reply:
This was how one of the other validate methods was doing it. It is returning the result from the validationErrorFromResult into a FormValidationResult, which all the fields will be after their own validations.